classes/ActionScheduler_ActionFactory.php

```diff
@@ -57,76 +59,141 @@ public function get_stored_action( $status, $hook, array $args = array(), Action
+	 * @param string $group A group to put the action in.
+	 *
+	 * @return int The ID of the stored action.
+	 */
+	public function recurring_unique( $hook, $unique, $args = array(), $first = null, $interval = null, $group = '' ) {
+		if ( empty( $interval ) ) {
+			return $this->single_unique( $hook, $unique, $args, $first, $group );
 		}
-		$date = as_get_datetime_object( $first );
+		$date     = as_get_datetime_object( $first );
 		$schedule = new ActionScheduler_IntervalSchedule( $date, $interval );
-		$action = new ActionScheduler_Action( $hook, $args, $schedule, $group );
-		return $this->store( $action );
+		$action   = new ActionScheduler_Action( $hook, $args, $schedule, $group );
+		return $unique ? $this->store_unique_action( $action ) : $this->store( $action );
 	}
 
 	/**
 	 * Create the first instance of an action recurring on a Cron schedule.
 	 *
-	 * @param string $hook The hook to trigger when this action runs
-	 * @param array $args Args to pass when the hook is triggered
-	 * @param int $base_timestamp The first instance of the action will be scheduled
+	 * @param string $hook The hook to trigger when this action runs.
+	 * @param array  $args Args to pass when the hook is triggered.
+	 * @param int    $base_timestamp The first instance of the action will be scheduled
 	 *        to run at a time calculated after this timestamp matching the cron
 	 *        expression. This can be used to delay the first instance of the action.
-	 * @param int $schedule A cron definition string
-	 * @param string $group A group to put the action in
+	 * @param int    $schedule A cron definition string.
+	 * @param string $group A group to put the action in.
 	 *
-	 * @return int The ID of the stored action
+	 * @return int The ID of the stored action.
 	 */
 	public function cron( $hook, $args = array(), $base_timestamp = null, $schedule = null, $group = '' ) {
-		if ( empty($schedule) ) {
-			return $this->single( $hook, $args, $base_timestamp, $group );
+		return $this->cron_unique( $hook, false, $args, $base_timestamp, $schedule, $group );
+	}
+
+	/**
+	 * Create the first instance of an action recurring on a Cron schedule only if there is no pending or running action with same name and params.
+	 *
+	 * @param string $hook The hook to trigger when this action runs.
+	 * @param bool   $unique Whether action scheduled should be unique.
+	 * @param array  $args Args to pass when the hook is triggered.
+	 * @param int    $base_timestamp The first instance of the action will be scheduled
+	 *        to run at a time calculated after this timestamp matching the cron
+	 *        expression. This can be used to delay the first instance of the action.
+	 * @param int    $schedule A cron definition string.
+	 * @param string $group A group to put the action in.
+	 *
+	 * @return int The ID of the stored action.
+	 **/
+	public function cron_unique( $hook, $unique, $args = array(), $base_timestamp = null, $schedule = null, $group = '' ) {
```

Comment:
Optional thing, just wanted to drop here for consideration (and applies to all of the new `*_unique()` methods):

Bearing in mind this method is public, if I was calling it directly I'd probably assume that (outside of special cases) I don't need to specify `$unique` is `true` (and the method description implies the same thing). With those things in mind, would:

```php
public function cron_unique( $hook, $args = array(), $base_timestamp = null, $schedule = null, $group = '', $unique = true )
```

Be a better signature (that is, make `$unique` optional and have it default to true)? Of course, changing this would also require updates to the matching non-unique methods.

Reply:
Hmm you are right, let me take care of this.

Reply:
Note that this was addressed in https://github.com/woocommerce/action-scheduler/pull/831/commits/c8652fe2344ebe2d264f64f7f8ad9874b33161a5